### PR TITLE
Add Transform property for Brush class

### LINKED ISSUE #5768: Add Transform property to Brushes
**Is your feature request related to a problem? Please describe.**
Allow brushes to be transformed, which WPF, UWP, and WinUI all support
WPF:
https://docs.microsoft.com/en-us/dotnet/desktop/wpf/graphics-multimedia/brush-transformation-overview?view=netframeworkdesktop-4.8

WinUI/UWP have a `Transform` and `RelativeTransform` property directly on `Brush` 
https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.media.brush?view=winrt-19041

I've been trying to port the new WinUI styles and they have a couple brushes that have scale transform applied:
````
 <LinearGradientBrush x:Key="TextControlElevationBorderBrush" MappingMode="Absolute" StartPoint="0,0" EndPoint="0,2">
                <LinearGradientBrush.RelativeTransform>
                    <ScaleTransform ScaleY="-1" CenterY="0.5"/>
                </LinearGradientBrush.RelativeTransform>
                <LinearGradientBrush.GradientStops>
                    <GradientStop Offset="0.5" Color="{StaticResource ControlStrongStrokeColorDefault}"/>
                    <GradientStop Offset="1.0" Color="{StaticResource ControlStrokeColorDefault}"/>
                </LinearGradientBrush.GradientStops>
</LinearGradientBrush>
````

These brushes are usually used for borders (from what I can tell) so a work around is to separate the border brush into a separate Border element and apply the transform there, but that's not as ideal

**Describe the solution you'd like**
I think either adding a Transform property to `IBrush` would work, or use a separate `ITransformableBrush` interface (naming for example) to just apply to Gradient/Image/etc. brushes since `SolidColorBrush` probably doesn't need this.

### LINKED ISSUE #6343: Add Transform and RelativeTransform properties for Brush'es
**Is your feature request related to a problem? Please describe.**

Add support for `Transform` and `RelativeTransform` (not sure if this one is needed) property for Brush'es (LinearGradientBrush, RadialGradientBrush and VisualBrush).

- [Brush.Transform Property](https://docs.microsoft.com/en-us/dotnet/api/system.windows.media.brush.transform?view=net-5.0)
- [Brush Transformation Overview](https://docs.microsoft.com/en-us/dotnet/desktop/wpf/graphics-multimedia/brush-transformation-overview?view=netframeworkdesktop-4.8)

**Describe the solution you'd like**

Add to Brush  two transformation properties: `Transform` and `RelativeTransform`. The properties enable rotate, scale, skew, and translate a brush's contents.

**Describe alternatives you've considered**

Did try manually transform brush properties, but that doesn't really work.

**Additional context**

It is very useful for example when converting Svg paint servers to Avalonia brushes. As paint server can use user space coordinates and bounding box coordinates and can have transform property set for brush transformations. Right now I am using LinearGradientBrush, RadialGradientBrush and VisualBrush and all of them require transform.

SkiaSharp support local matrix parameter when creating shaders for brushes:

- [CreateLinearGradient](https://docs.microsoft.com/en-us/dotnet/api/skiasharp.skshader.createlineargradient?view=skiasharp-2.80.2#SkiaSharp_SKShader_CreateLinearGradient_SkiaSharp_SKPoint_SkiaSharp_SKPoint_SkiaSharp_SKColorF___SkiaSharp_SKColorSpace_System_Single___SkiaSharp_SKShaderTileMode_SkiaSharp_SKMatrix_)

- [CreateRadialGradient](https://docs.microsoft.com/en-us/dotnet/api/skiasharp.skshader.createradialgradient?view=skiasharp-2.80.2#SkiaSharp_SKShader_CreateRadialGradient_SkiaSharp_SKPoint_System_Single_SkiaSharp_SKColorF___SkiaSharp_SKColorSpace_System_Single___SkiaSharp_SKShaderTileMode_SkiaSharp_SKMatrix_)

- [CreateTwoPointConicalGradient](https://docs.microsoft.com/en-us/dotnet/api/skiasharp.skshader.createtwopointconicalgradient?view=skiasharp-2.80.2#SkiaSharp_SKShader_CreateTwoPointConicalGradient_SkiaSharp_SKPoint_System_Single_SkiaSharp_SKPoint_System_Single_SkiaSharp_SKColor___SkiaSharp_SKShaderTileMode_)

- [CreatePicture](https://docs.microsoft.com/en-us/dotnet/api/skiasharp.skshader.createpicture?view=skiasharp-2.80.2#SkiaSharp_SKShader_CreatePicture_SkiaSharp_SKPicture_SkiaSharp_SKShaderTileMode_SkiaSharp_SKShaderTileMode_SkiaSharp_SKMatrix_SkiaSharp_SKRect_)

That is what I use for Svg rendering.

I am converting the svg brushes to WPF xaml here https://github.com/wieslawsoltes/SvgGradientsDemo for reference.

### PR DESCRIPTION
**Pull request by wieslawsoltes**
## Checklist

- [ ] Added unit tests (if possible)?
- [ ] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Documentation with user documentation

## Fixed issues

Fixes: #6343
Fixes: #5768

### CODE REVIEW

**Comment by wieslawsoltes** [comment]
image
(Left-Top: Skia, Left-Bottom: WPF, Right: Avalonia)

Avalonia SandBox xaml:
```XAML
    <Canvas Background="White" Width="480" Height="360">
        <Rectangle Canvas.Left="20" Canvas.Top="20" Width="440" Height="50">
            <Rectangle.Fill>
                <!-- gradientUnits="userSpaceOnUse" x1="0" y1="0" x2="430" y2="0" gradientTransform="translate(25 35) scale(0.5)" -->
                <!-- NOTE: Asjust StartPoint and EndPoint to -Canvas.Left and -Canvas.Top -->
                <LinearGradientBrush StartPoint="0,0" EndPoint="410,0" >
                    <LinearGradientBrush.Transform>
                        <TransformGroup>
                            <ScaleTransform ScaleX="0.5"/>
                            <SkewTransform/>
                            <RotateTransform/>
                            <TranslateTransform X="5" Y="15"/> <!-- NOTE: Asjust  X and Y to -Canvas.Left and -Canvas.Top -->
                        </TransformGroup>
                    </LinearGradientBrush.Transform>
                    <LinearGradientBrush.GradientStops>
                        <GradientStop Color="Blue" Offset="0"/>
                        <GradientStop Color="Green"  Offset="0.5"/>
                        <GradientStop Color="Lime" Offset="1" />
                    </LinearGradientBrush.GradientStops>
                </LinearGradientBrush>
            </Rectangle.Fill>
        </Rectangle>
        <TextBlock Canvas.Left="20" Canvas.Top="70" FontSize="30" Text="scale(0.5) on gradient"/>
        <Rectangle Canvas.Left="20" Canvas.Top="110" Width="440" Height="50">
            <Rectangle.Fill>
                <!-- gradientUnits="userSpaceOnUse" cx="0" cy="0" fx="0" fy="0" r="60" gradientTransform="translate(240,155) skewX(45)" -->
                <!-- NOTE: Radius is always relative to convert: Radius / Width = 60 / 440 = 0.13636364 --> 
                <RadialGradientBrush Center="0.0,0.0" GradientOrigin="0.0,0.0" Radius="0.13636364">
                    <RadialGradientBrush.Transform>
                        <TransformGroup>
                            <ScaleTransform/>
                            <SkewTransform AngleX="45"/>
                            <RotateTransform/>
                            <TranslateTransform X="240" Y="45"/> <!-- NOTE: Asjust X and Y to -Canvas.Left and -Canvas.Top -->
                        </TransformGroup>
                    </RadialGradientBrush.Transform>
                    <RadialGradientBrush.GradientStops>
                        <GradientStop Color="Black" Offset="0"/>
                        <GradientStop Color="#FFFFA500" Offset="1"/>
                    </RadialGradientBrush.GradientStops>
                </RadialGradientBrush>
            </Rectangle.Fill>
        </Rectangle>
        <TextBlock Canvas.Left="20" Canvas.Top="160" FontSize="30" Text="skewX(45) on gradient"/>
        <Rectangle Canvas.Left="20" Canvas.Top="210" Width="440" Height="50">
            <Rectangle.Fill>
                <!-- x="0" y="0" width="20" height="20" patternTransform="translate(25 215) scale(2) skewX(45)" -->
                <VisualBrush TileMode="Tile" SourceRect="0,0,20,20" DestinationRect="0,0,20,20" Stretch="None">
                    <VisualBrush.Transform>
                        <TransformGroup>
                            <ScaleTransform ScaleX="2" ScaleY="2"/>
                            <SkewTransform AngleX="45"/>
                            <RotateTransform/>
                            <TranslateTransform X="5" Y="5"/> <!-- NOTE: Asjust X and Y to -Canvas.Left and -Canvas.Top -->
                        </TransformGroup>
                    </VisualBrush.Transform>
                    <VisualBrush.Visual>
                        <Canvas Width="20" Height="20">
                            <Rectangle Canvas.Left="0" Canvas.Top="0" Width="10" Height="10" Fill="Maroon"/>
                            <Rectangle Canvas.Left="10" Canvas.Top="0" Width="10" Height="10" Fill="Green"/>
                            <Rectangle Canvas.Left="0" Canvas.Top="10" Width="10" Height="10" Fill="Blue"/>
                            <Rectangle Canvas.Left="10" Canvas.Top="10" Width="10" Height="10" Fill="Yellow"/>
                        </Canvas>
                    </VisualBrush.Visual>
                </VisualBrush>
            </Rectangle.Fill>
        </Rectangle>
        <TextBlock Canvas.Left="20" Canvas.Top="260" FontSize="30" Text="scale(2), skewX(45) on pattern"/>
    </Canvas>
```

WPF xaml:
```xaml
    <Canvas Background="White" Width="480" Height="360">
        <Rectangle Canvas.Left="20" Canvas.Top="20" Width="440" Height="50">
            <Rectangle.Fill>
                <!-- gradientUnits="userSpaceOnUse" x1="0" y1="0" x2="430" y2="0" gradientTransform="translate(25 35) scale(0.5)" -->
                <!-- NOTE: Asjust StartPoint and EndPoint to -Canvas.Left and -Canvas.Top -->
                <LinearGradientBrush StartPoint="0,0" EndPoint="410,0" MappingMode="Absolute">
                    <LinearGradientBrush.Transform>
                        <TransformGroup>
                            <ScaleTransform CenterY="0.5" CenterX="0.5" ScaleX="0.5"/>
                            <SkewTransform CenterY="0.5" CenterX="0.5"/>
                            <RotateTransform CenterY="0.5" CenterX="0.5"/>
                            <TranslateTransform X="5" Y="15"/> <!-- NOTE: Asjust  X and Y to -Canvas.Left and -Canvas.Top -->
                        </TransformGroup>
                    </LinearGradientBrush.Transform>
                    <LinearGradientBrush.GradientStops>
                        <GradientStop Color="Blue" Offset="0"/>
                        <GradientStop Color="Green"  Offset="0.5"/>
                        <GradientStop Color="Lime" Offset="1" />
                    </LinearGradientBrush.GradientStops>
                </LinearGradientBrush>
            </Rectangle.Fill>
        </Rectangle>
        <TextBlock Canvas.Left="20" Canvas.Top="70" FontSize="30" Text="scale(0.5) on gradient"/>
        <Rectangle Canvas.Left="20" Canvas.Top="110" Width="440" Height="50">
            <Rectangle.Fill>
                <!-- gradientUnits="userSpaceOnUse" cx="0" cy="0" fx="0" fy="0" r="60" gradientTransform="translate(240,155) skewX(45)" -->
                <RadialGradientBrush Center="0.0,0.0" GradientOrigin="0.0,0.0" RadiusX="60" RadiusY="60" MappingMode="Absolute">
                    <RadialGradientBrush.Transform>
                        <TransformGroup>
                            <ScaleTransform CenterY="0.5" CenterX="0.5"/>
                            <SkewTransform CenterY="0.5" CenterX="0.5" AngleX="45"/>
                            <RotateTransform CenterY="0.5" CenterX="0.5"/>
                            <TranslateTransform X="240" Y="45"/> <!-- NOTE: Asjust X and Y to -Canvas.Left and -Canvas.Top -->
                        </TransformGroup>
                    </RadialGradientBrush.Transform>
                    <RadialGradientBrush.GradientStops>
                        <GradientStop Color="Black" Offset="0"/>
                        <GradientStop Color="#FFFFA500" Offset="1"/>
                    </RadialGradientBrush.GradientStops>
                </RadialGradientBrush>
            </Rectangle.Fill>
        </Rectangle>
        <TextBlock Canvas.Left="20" Canvas.Top="160" FontSize="30" Text="skewX(45) on gradient"/>
        <Rectangle Canvas.Left="20" Canvas.Top="210" Width="440" Height="50">
            <Rectangle.Fill>
                <!-- x="0" y="0" width="20" height="20" patternTransform="translate(25 215) scale(2) skewX(45)" -->
                <VisualBrush TileMode="Tile" Viewport="0,0,20,20" ViewportUnits="Absolute" Viewbox="0,0,20,20" ViewboxUnits="Absolute" Stretch="None">
                    <VisualBrush.Transform>
                        <TransformGroup>
                            <ScaleTransform CenterY="0.5" CenterX="0.5" ScaleX="2" ScaleY="2"/>
                            <SkewTransform CenterY="0.5" CenterX="0.5" AngleX="45"/>
                            <RotateTransform CenterY="0.5" CenterX="0.5"/>
                            <TranslateTransform X="5" Y="5"/> <!-- NOTE: Asjust X and Y to -Canvas.Left and -Canvas.Top -->
                        </TransformGroup>
                    </VisualBrush.Transform>
                    <VisualBrush.Visual>
                        <Canvas Width="20" Height="20">
                            <Rectangle Canvas.Left="0" Canvas.Top="0" Width="10" Height="10" Fill="Maroon"/>
                            <Rectangle Canvas.Left="10" Canvas.Top="0" Width="10" Height="10" Fill="Green"/>
                            <Rectangle Canvas.Left="0" Canvas.Top="10" Width="10" Height="10" Fill="Blue"/>
                            <Rectangle Canvas.Left="10" Canvas.Top="10" Width="10" Height="10" Fill="Yellow"/>
                        </Canvas>
                    </VisualBrush.Visual>
                </VisualBrush>
            </Rectangle.Fill>
        </Rectangle>
        <TextBlock Canvas.Left="20" Canvas.Top="260" FontSize="30" Text="scale(2), skewX(45) on pattern"/>
    </Canvas>
````

Original svg:
```svg
<svg version="1.1" baseProfile="basic" id="svg-root"
  width="100%" height="100%" viewBox="0 0 480 360"
  xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
  <!--======================================================================-->
  <!--=  SVG 1.1 2nd Edition Test Case                                     =-->
  <!--======================================================================-->
  <!--=  Copyright 2009 World Wide Web Consortium, (Massachusetts          =-->
  <!--=  Institute of Technology, European Research Consortium for         =-->
  <!--=  Informatics and Mathematics (ERCIM), Keio University).            =-->
  <!--=  All Rights Reserved.                                              =-->
  <!--=  See http://www.w3.org/Consortium/Legal/.                          =-->
  <!--======================================================================-->
  <d:SVGTestCase xmlns:d="http://www.w3.org/2000/02/svg/testsuite/description/"
    template-version="1.4" reviewer="SVGWG" author="Haroon Sheikh" status="accepted"
    version="$Revision: 1.9 $" testname="$RCSfile: pservers-grad-06-b.svg,v $">
    <d:testDescription xmlns="http://www.w3.org/1999/xhtml" href="http://www.w3.org/TR/SVG11/pservers.html#Gradients">
      <p>
        Test that the viewer can handle the gradientTransform and the patternTransform
        attribute on gradients and patterns respectively.
      </p>
      <p>
        From top-down the appearance of objects is as follows.
      </p>
      <p>
        The top rectangle has a linear gradient whose coordinate system has been scaled down by
        a half. So the gradient travelling from left to right (from blue to green to lime) should
        only occuply the left half the rectangle.
      </p>
      <p>
        The next rectangle has radial gradient that has been translated to the center and skewed
        in the positive X direction by 45 degrees. Therefore the gradient should appear
        ellipltical and rotated around the center.
      </p>
      <p>
        The last row contains a rectangle with pattern on the fill. The transformation on the
        pattern moves the coordinate system to the top left of the rectangle and then scales it
        by a factor of 2 and then skew's it in the X direction by 45 degrees. The pattern
        consists of a 2 by 2 array of colored rectangles.
      </p>
    </d:testDescription>
    <d:operatorScript xmlns="http://www.w3.org/1999/xhtml">            
      <p>Run the test. No interaction required.</p>
    </d:operatorScript>
    <d:passCriteria xmlns="http://www.w3.org/1999/xhtml">
      <p>The test passes if the rendering matches the reference image, apart
      from any differences in font choice due to CSS2 rules.  Specifically:</p>
      <ul>
        <li>The top rectangle is filled with a linear gradient from blue on the left,
        to lime in the middle.  The right half of the rectangle is filled with plain lime.</li>
        <li>The middle rectangle is filled with an elliptical radial gradient with
        black on the inside and orange on the outside.  The center point of the gradient
        is near the center-bottom of the rectangle.  The gradient is skewed, so that it appears
        as a rotated elliptical gradient.</li>
        <li>The bottom rectangle is filled with a repeating pattern of tiled
        red, green, yellow and blue parallelograms.</li>
      </ul>
    </d:passCriteria>
  </d:SVGTestCase>
  <title id="test-title">$RCSfile: pservers-grad-06-b.svg,v $</title>
  <defs>
    <font-face font-family="SVGFreeSansASCII" unicode-range="U+0-7F">
      <font-face-src>
        <font-face-uri xlink:href="../resources/SVGFreeSans.svg#ascii"/>
      </font-face-src>
    </font-face>
  </defs>
  <g id="test-body-content" font-family="SVGFreeSansASCII,sans-serif" font-size="18">
    <!-- ====================================================================== -->
    <!-- Linear Gradient with gradientTransforms                                -->
    <!-- ====================================================================== -->
    <linearGradient id="Grad1" gradientUnits="userSpaceOnUse" x1="0" y1="0" x2="430" y2="0" gradientTransform="translate(25 35) scale(0.5)">
      <stop stop-color="blue" offset="0"/>
      <stop stop-color="green" offset="0.5"/>
      <stop stop-color="lime" offset="1"/>
    </linearGradient>
    <rect x="20" y="20" width="440" height="50" fill="url(#Grad1)"/>
    <text font-size="30" x="20" y="100">scale(0.5) on gradient</text>
    <!-- ====================================================================== -->
    <!-- Radial Gradiant with a gradientTransform                               -->
    <!-- ====================================================================== -->
    <radialGradient id="Grad2" gradientUnits="userSpaceOnUse" cx="0" cy="0" fx="0" fy="0" r="60" gradientTransform="translate(240,155) skewX(45)">
      <stop stop-color="black" offset="0"/>
      <stop stop-color="rgb(255,165,0)" offset="1"/>
    </radialGradient>
    <rect x="20" y="110" width="440" height="50" fill="url(#Grad2)"/>
    <text font-size="30" x="20" y="190">skewX(45) on gradient</text>
    <!-- ====================================================================== -->
    <!-- Pattern filled rectangles with a patternTransform.                     -->
    <!-- ====================================================================== -->
    <pattern patternUnits="userSpaceOnUse" id="Pat3" x="0" y="0" width="20" height="20" patternTransform="translate(25 215) scale(2) skewX(45)">
      <rect x="0" y="0" width="10" height="10" fill="maroon"/>
      <rect x="10" y="0" width="10" height="10" fill="green"/>
      <rect x="0" y="10" width="10" height="10" fill="blue"/>
      <rect x="10" y="10" width="10" height="10" fill="yellow"/>
    </pattern>
    <rect x="20" y="210" width="440" height="50" fill="url(#Pat3)"/>
    <text font-size="30" x="20" y="290">scale(2), skewX(45) on pattern</text>
  </g>
  <g font-family="SVGFreeSansASCII,sans-serif" font-size="32">
    <text id="revision" x="10" y="340" stroke="none" fill="black">$Revision: 1.9 $</text>
  </g>
  <rect id="test-frame" x="1" y="1" width="478" height="358" fill="none" stroke="#000000"/>
  <!-- comment out this watermark once the test is approved -->
  <!--<g id="draft-watermark">
    <rect x="1" y="1" width="478" height="20" fill="red" stroke="black" stroke-width="1"/>
    <text font-family="SVGFreeSansASCII,sans-serif" font-weight="bold" font-size="20" x="240"
      text-anchor="middle" y="18" stroke-width="0.5" stroke="black" fill="white">DRAFT</text>
  </g>-->
</svg>

```

File: src/Avalonia.Visuals/Media/Brush.cs [hunk 1]
```diff
@@ -18,13 +18,19 @@ public abstract class Brush : Animatable, IMutableBrush
         public static readonly StyledProperty<double> OpacityProperty =
             AvaloniaProperty.Register<Brush, double>(nameof(Opacity), 1.0);
 
+        /// <summary>
+        /// Defines the <see cref="Transform"/> property.
+        /// </summary>
+        public static readonly StyledProperty<Transform> TransformProperty =
```
Comment (maxkatz6) [inline src/Avalonia.Visuals/Media/Brush.cs:24]: We have ITransform interface, which is used on Control (but not on Geometry).
Should we use it instead?

Comment (wieslawsoltes) [reply src/Avalonia.Visuals/Media/Brush.cs]: I am converting type from Transform to ITransform as it needs to be converted to immutable for deffered renderer.

Comment (wieslawsoltes) [reply src/Avalonia.Visuals/Media/Brush.cs]: I have changed to ITransform and introduced immutable transform to fix deferred rendering

File: src/Avalonia.Visuals/Animation/Animators/GradientBrushAnimator.cs [hunk 2]
```diff
@@ -30,6 +30,7 @@ public class GradientBrushAnimator : Animator<IGradientBrush?>
                     return new ImmutableRadialGradientBrush(
                         InterpolateStops(progress, oldValue.GradientStops, newValue.GradientStops),
                         s_doubleAnimator.Interpolate(progress, oldValue.Opacity, newValue.Opacity),
+                        oldValue.Transform,
```
Comment (maxkatz6) [inline src/Avalonia.Visuals/Animation/Animators/GradientBrushAnimator.cs:33]: TransformAnimator should be reused or refactored, so we can use Interpolate from there.
Probably should be done in another PR, if it's not straightforward.

Comment (wieslawsoltes) [reply src/Avalonia.Visuals/Animation/Animators/GradientBrushAnimator.cs]: I think separate PR or maybe @jmacato can give some input here

Comment (wieslawsoltes) [reply src/Avalonia.Visuals/Animation/Animators/GradientBrushAnimator.cs]: I have added TransformAnimator stub method.

Comment (wieslawsoltes) [reply src/Avalonia.Visuals/Animation/Animators/GradientBrushAnimator.cs]: Add implementation to TransformAnimator

**Comment by wieslawsoltes** [comment]
Another usage example is animations

```xaml
<Border Width="350" Height="350" Child="{x:Null}">
  <Border.Styles>
    <Style Selector="Border">
      <Style.Animations>
        <Animation Duration="0:0:5" IterationCount="Infinite" PlaybackDirection="Normal">
          <KeyFrame Cue="0%">
            <Setter Property="Background">
              <RadialGradientBrush Center="50%,50%" Radius="1.0">
                <RadialGradientBrush.Transform>
                  <TransformGroup>
                    <ScaleTransform />
                    <SkewTransform />
                    <RotateTransform />
                    <TranslateTransform X="-500" Y="0" />
                  </TransformGroup>
                </RadialGradientBrush.Transform>
                <GradientStop Offset="0" Color="Red"/>
                <GradientStop Offset="1" Color="Blue"/>
              </RadialGradientBrush>
            </Setter>
          </KeyFrame>
          <KeyFrame Cue="50%">
            <Setter Property="Background">
              <RadialGradientBrush Center="50%,50%" Radius="0.1">
                <RadialGradientBrush.Transform>
                  <TransformGroup>
                    <ScaleTransform />
                    <SkewTransform />
                    <RotateTransform />
                    <TranslateTransform X="0" Y="0" />
                  </TransformGroup>
                </RadialGradientBrush.Transform>
                <GradientStop Offset="0" Color="Red"/>
                <GradientStop Offset="1" Color="Blue"/>
              </RadialGradientBrush>
            </Setter>
          </KeyFrame>
          <KeyFrame Cue="100%">
            <Setter Property="Background">
              <RadialGradientBrush Center="50%,50%" Radius="1.0">
                <RadialGradientBrush.Transform>
                  <TransformGroup>
                    <ScaleTransform />
                    <SkewTransform />
                    <RotateTransform />
                    <TranslateTransform X="500" Y="0" />
                  </TransformGroup>
                </RadialGradientBrush.Transform>
                <GradientStop Offset="0" Color="Red"/>
                <GradientStop Offset="1" Color="Blue"/>
              </RadialGradientBrush>
            </Setter>
          </KeyFrame>
        </Animation>
      </Style.Animations>
    </Style>
  </Border.Styles>
</Border>
```

https://user-images.githubusercontent.com/2297442/127819647-b4085325-19df-4a07-8f11-8fb6b2e41830.mp4

File: src/Avalonia.Visuals/Animation/Animators/GradientBrushAnimator.cs [hunk 3]
```diff
@@ -86,23 +91,31 @@ private IReadOnlyList<ImmutableGradientStop> InterpolateStops(double progress, I
             return stops;
         }
 
+        private ImmutableTransform InterpolateTransform(double progress, ITransform oldValue, ITransform newValue)
+        {
+            return new ImmutableTransform(_matrixAnimator.Interpolate(progress, oldValue.Value, newValue.Value));
```
Comment (maxkatz6) [inline src/Avalonia.Visuals/Animation/Animators/GradientBrushAnimator.cs:96]: I wonder why for brush transform we can use way simpler matrix interpolation, while for control transform operations, we have way more complex implementation (that does not interpolate resulting matrix) - https://github.com/AvaloniaUI/Avalonia/blob/81e49c329e5b8461a780c41e7be525aaae0b570b/src/Avalonia.Visuals/Media/Transformation/TransformOperations.cs#L84

Is matrix interpolation less accurate? CC @MarchingCube

Comment (MarchingCube) [reply src/Avalonia.Visuals/Animation/Animators/GradientBrushAnimator.cs:96]: I think that simply interpolating values of transform will not produce desired result (it might work for simplest transforms but for more complex ones it won't look good). Interpolation in control transform is implemented to follow css spec and retain order of operations if possible.

Comment (MarchingCube) [reply src/Avalonia.Visuals/Animation/Animators/GradientBrushAnimator.cs:96]: Animation

Example of rotation from identity to a rotation of PI. Left is matrix interpolation from this PR, right one is transform operations interpolation.

**Comment by grokys** [comment]
What's the status on this? I've not been involved in reviewing it; have the questions been resolved?

**Comment by wieslawsoltes** [comment]
> What's the status on this? I've not been involved in reviewing it; have the questions been resolved?

@grokys Did not address the questions yet, also implementation needs more testing.

**Comment by desneill** [comment]
It would be great to have this feature added.  Do we know when it might be ready?   I would like to use the VisualBrush to draw lines rather than just tiling a background

**Comment by maxkatz6** [comment]
@wieslawsoltes we can leave animation problem for a future, and merge main feature first

**Comment by MarchingCube** [comment]
I agree with @maxkatz6, base feature looks fine but animation/transition needs more work that can be done later. One cannot simply interpolate raw matrix values since that does not work for rotations. It works for translation, scale and skew only. Also CSS spec is generally suggesting to do a decomposed matrix interpolation instead since then standard transforms will generally interpolate correctly. https://www.w3.org/TR/css-transforms-1/#interpolation-of-2d-matrices

**Comment by jmacato** [comment]
Closing due to inactivity and merge conflicts

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0019215-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Comment by wieslawsoltes** [comment]
Removed transform animation and fixed merge conflicts.

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0019218-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Review by maxkatz6 — APPROVED** [review]
LGTM!
